util/headerreader/header_reader.go

```diff
@@ -176,6 +180,12 @@ func (s *HeaderReader) possiblyBroadcast(h *types.Header) {
 	}
 }
 
+func (s *HeaderReader) setError(err error) {
```

Comment:
I'm not sure I understand the usecase, but there probably should be somewhere a call to setError(nil) in case we had a transient error that's now over.

Reply:
We set `lastBroadcastErr` to `nil` in `possiblyBroadcast`, but this function also grabs the lock which we already have in `possiblyBroadcast`